sisl/geom/nanoribbon.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+
+from sisl import geom, Atom, Geometry
+
+__all__ = ['nanoribbon']
+
+
+def nanoribbon(width, kind='armchair', bond=1.42, atom=None, sat_bond=1.09, sat_atom=None):
+    r""" Construction of a nanoribbon unit cell of type armchair or zigzag.
+    The geometry is oriented along the x-axis.
+
+    Parameters
+    ----------
+    width : int
+       number of atoms in the transverse direction
+    kind : {'armchair', 'zigzag'}
+       type of ribbon
+    bond : float, optional
+       bond length between atoms in the honeycomb lattice. Defaults to 1.42
+    atom : Atom, optional
+       atom (or atoms) in the honeycomb lattice. Defaults to ``Atom(6)``
+    sat_bond : float, optional
+       bond length to the edge saturation atoms. A negative value will return 
+       a nanoribbon without edge decoration. Defaults to 1.09
+    sat_atom : Atom, optional
+       atom (or atoms) for the edge decoration. Defaults to ``Atom(1)``
+
+    See Also
+    --------
+    honeycomb: honeycomb lattices
+    graphene: graphene geometry
+    """
+    if atom is None:
+        atom = Atom(Z=6, R=bond * 1.01)
+    if sat_atom is None:
+        sat_atom = Atom(Z=1, R=sat_bond * 1.01)
+
+    # Width characterization
+    if not isinstance(width, int):
+        raise ValueError("nanoribbon: the width needs to be a postive integer!")
+    width = max(width, 1)
+    n, m = width // 2, width % 2
+
+    ribbon = geom.honeycomb(bond, atom, orthogonal=True)
+
+    kind = kind.lower()
+    if kind.startswith('a'):
```

Comment:
probably we should force `kind=='armchair'` to be sure users do it correctly.
For instance with the nano-porous graphene GNR the `kind` could also be added.

Actually one could add an optional parameter `length=1` which automatically tiles along length of ribbon. Then `width` and `length` could be tuples/lists where one appends the nanoribbons of those individual ribbons together? That would make one able to do rather complicated nano-ribbons with very little complexity (append would work seamlessly and a nested call would be really easy ;)

What do you think?
If

Reply:
Hmm, I think my snippet below actually takes care of all cases? I'll give you a more detailed example once done.

Reply:
OK, I will take a closer look

Reply:
No need to, unless you want, I have a working solution for armchair, now generalizing for zig-zag :)